src/DynamoApplications/StartupUtils.cs

```diff
@@ -273,10 +275,21 @@ public static DynamoModel MakeModel(bool CLImode, string asmPath = "", string ho
         /// <returns></returns>
         public static DynamoModel MakeModel(bool CLImode, string asmPath = "", HostAnalyticsInfo info = new HostAnalyticsInfo())
         {
+            var pathManagerAndPreference = CreatePathManagerAndPreference();            
+            DynamoModel.OnDetectLanguage();
+
             // Preload ASM and display corresponding message on splash screen
             DynamoModel.OnRequestUpdateLoadBarStatus(new SplashScreenLoadEventArgs(Resources.SplashScreenPreLoadingAsm, 10));
             var isASMloaded = PreloadASM(asmPath, out string geometryFactoryPath, out string preloaderLocation);
-            var model = StartDynamoWithDefaultConfig(CLImode, string.Empty, string.Empty, geometryFactoryPath, preloaderLocation, info);
+            var model = StartDynamoWithDefaultConfig(CLImode,
+                string.Empty,
+                string.Empty,
+                geometryFactoryPath,
+                preloaderLocation,
+                info,
+                false,
+                pathManagerAndPreference.Item1,
+                pathManagerAndPreference.Item2);
```

Comment:
Passing the items does not count as singleton design pattern. Please refer to https://github.com/DynamoDS/Dynamo/blob/master/src/NodeServices/PythonServices.cs#L113